content/migrate/tf2pulumi.md

```diff
@@ -0,0 +1,420 @@
+            $("pulumi-chooser[type='language'] > ul > li > a").css({ "cursor": "pointer" });
+        });
+}
+
+// downloadCode downloads the currently converted code, if available.
+function downloadCode() {
+    let zip = new JSZip();
+    for (let fn of Object.keys(currentCodeFiles)) {
+        zip.file(fn, currentCodeFiles[fn]);
+    }
+    zip.generateAsync({ type: "blob" }).then(function(content) {
+        // Use FileSaver.js to save the file, triggering download in the user's browser.
+        saveAs(content, "tf2pulumi.zip");
+    });
+}
+
+window.onload = function() {
+    $(document).ready(function() {
+        // If there are querystring parameters populate the fields.
+        let tfUrl = getQueryVariable("url");
+        let tfCode = getQueryVariable("code");
+        if (tfUrl) {
+            $("#terraform-url").val(tfUrl);
+            setInputKind("url");
+        } else {
+            if (!tfCode) {
+                // Populate a simple default example.
+                let comment = "#"; // to suppress Markdown lint errors.
+                tfCode = `${comment} This Terraform sample provisions an AWS EC2 instance running Ubuntu.
+${comment} Choose a language and click CONVERT below -- or try replacing it with your own!
+
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
+
+resource "aws_instance" "web" {
+  ami           = "\${data.aws_ami.ubuntu.id}"
+  instance_type = "t2.micro"
+
+  tags = {
+    Name = "HelloWorld"
+  }
+}
+`;
+            }
+            $("#terraform-code").val(tfCode);
+            setInputKind("code");
+        }
+
+        // Clear the URL box when selecting the code box.
+        $("#terraform-code").click(function() {
+            $(this).select()
+            $("#terraform-url").val("https://");
+            $("#terraform-upload").val("");
+        });
+
+        // URL is pre-populated with "https://" -- select it upon click.
+        // Also clear the code box since we can't issue multiple at once.
+        $("#terraform-url").click(function() {
+            $(this).select();
+            $("#terraform-code").val("");
+            $("#terraform-upload").val("");
+        });
+
+        // Clear the other text boxes when selecting.
+        $("#terraform-upload").click(function() {
+            $("#terraform-code").val("");
+            $("#terraform-url").val("https://");
+        });
+
+        // If you hit enter in the URL bar, submit.
+        $("#terraform-url").keypress(function (e) {
+            if (e.which == 13) {
+                convertCode();
+                return false;
+            }
+        });
+
+        // Hook up event handlers for the language choosers.
+        $("pulumi-chooser[type='language'] > ul > li > a").each(function (i, e) {
```

Comment:
Technically okay, but I'd probably suggest just reading this from localStorage, which we update on every select change:

```
JSON.parse(localStorage.getItem("pulumi_state")).preferences.language
```

or similar should work.

Reply:
I'll give that a try. I still need the selector though, here, since I'm wiring up a new event handler, I think.